userprofile/views.py

```diff
@@ -111,6 +99,14 @@ def set_user_language(request):
     return response
 
 
+def regenerate_access_token(request):
+    """Regenerates the API access token."""
+    next = get_next_url(request)
```

Comment:
I don't understand why the `next` parameter is here. I thought that the profile page would include a button for regenerating the personal API token. Submitting the form sends HTTP POST to the server, which removes the token and sends a normal redirect response back to the browser. The user can be simply redirected back to the same profile page. This does not require any special `next` parameters in the URL or POST data for defining the redirect target.

Are you trying to use AJAX in the form submission? I can't see such code, but I didn't think this feature would need any AJAX anyway.

If the `next` parameter copied from the course language menu is not needed here, as I suspect, then it wasn't necessary to modify `set_user_lang()` or add `get_next_url()`.

Reply:
I didn't really understand the reason for having the ``next`` parameter in ``set_user_lang()``, and I used it as an example for ``regenerate_access_token()``. Therefore, I implemented it the same way with the ``next`` parameter, in case there was some reason it was needed. I'll remove it.